src/chocolatey.resources/helpers/functions/Get-EnvironmentVariable.ps1

```diff
@@ -10,11 +10,27 @@
 # distributed under the License is distributed on an "AS IS" BASIS,
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
-# limitations under the License.
-
-function Get-EnvironmentVariable([string] $Name, [System.EnvironmentVariableTarget] $Scope) {
-    [Environment]::GetEnvironmentVariable($Name, $Scope)
-}
-
-# Some enhancements to think about here.
-# $machinePath = [Microsoft.Win32.Registry]::LocalMachine.OpenSubKey("SYSTEM\CurrentControlSet\Control\Session Manager\Environment\").GetValue("PATH", "", [Microsoft.Win32.RegistryValueOptions]::DoNotExpandEnvironmentNames).ToString();
+# limitations under the License.
+
+function Get-EnvironmentVariable([string] $Name, [System.EnvironmentVariableTarget] $Scope, [bool] $PreserveVariables = $False) {

```

Comment:
It would be more elegant (and more conforming with PowerShell conventions) to declare the argument as [switch] instead of [bool].